doc/ui_component2_transition.md

```diff
@@ -0,0 +1,132 @@
+- For a component that reads a single prop and renders a simple div, upgrading it to UiComponent2 results in:
+  - initial renders that are 11% faster
+  - rerenders that are 6.5% faster
+- Special case: for a component that has simple lifecycles, but heavy prop reads, upgrading it to 
+UiComponent2 results in:
+  - initial renders that are 12% faster
+  - rerenders that are 28% faster
+  
+### JS-Backed Maps
+Props/state key-value pairs are stored directly on JS objects, instead of within Dart Maps which are opaque to React. Results in:
+
+- Modest overall rendering perf improvements due to decreased copying, garbage creation, and Dart map overhead
+- Props/state show up in the React DevTools, and primitive values (strings, numbers, booleans) can be edited live
+- Easier to interop with JS libraries that manipulate props, like react-redux
+- Easier to maintain react-dart/over_react interop and add new features
+
+### Lifecycle Method Changes
+
+- Drops support for unsafe lifecycle methods:
+  - componentWillMount
+  - componentWillReceiveProps
+  - componentWillUpdate
+- Adds support for new, safe lifecycle methods:
+  - getDerivedStateFromProps
+  - getSnapshotBeforeUpdate
+  - componentDidCatch / getDerivedStateFromError (adds support for [error boundaries](https://reactjs.org/docs/error-boundaries.html))
+- componentDidUpdate now receives a `snapshot` argument, which is the return value of getSnapshotBeforeUpdate.
+  
+See the [React docs](https://reactjs.org/docs/react-component.html#the-component-lifecycle) for information on why the old methods are unsafe and how to transition to the new ones. This 
+information is also available and linked to within the lifecycle methods' doc comments.
+
+### ErrorBoundary Component Added
+
+With the ability to utilize componentDidCatch / getDerivedStateFromError comes the ability to use [error boundaries](https://reactjs.org/docs/error-boundaries.html).
+A component (called ErrorBoundary) is present within OverReact and can be used to wrap a component tree to prevent it 
+from unmounting when an error is thrown. Alternatively, a custom error boundary can be created by utilizing the 
+[ErrorBoundaryMixin](../lib/src/component/error_boundary_mixins.dart).
+
+Usage of the default error boundary can be found in the Component1 examples (within web), and usage of a custom error 
+boundary can be found in the Component2 examples.
+
+### Other Method Changes
+- `getDefaultProps` is now `get defaultProps`
+- `getInitialState` is now `get initialState`
+- `..addProps(copyUnconsumedProps())` is now `..modifyProps(addUnconsumedProps)`. The same is true with `DomProps`, 
+the usage being `..modifyProps(addUnconsumedDomProps)`.
+
+### Prop validation goes through PropTypes
+- Shows "component stack" that helps determine where the error came from
+- React automatically de-duplicates the same warnings
+- __No longer throws, but warns__. If tests exist verifying that prop validation occurs, these will likely need to 
```

Comment:
We should mention that this only applies if they are update the component to UiComponent2